src/tool_doswin.c

```diff
@@ -702,6 +703,64 @@ struct curl_slist *GetLoadedModulePaths(void)
   return slist;
 }
 
+/* The terminal settings before enabling VT (Virtual Terminal) output */
+static struct TerminalSettings {
+  HANDLE hStdOut;
+  DWORD dwOutputMode;
+  LONG valid;
+} TerminalSettings;
+
+#ifndef ENABLE_VIRTUAL_TERMINAL_PROCESSING
+#define ENABLE_VIRTUAL_TERMINAL_PROCESSING 0x0004
+#endif
+
+static void restore_terminal(void)
+{
+  if(InterlockedExchange(&TerminalSettings.valid, (LONG)FALSE))
```

Comment:
Perhaps I am missing something, but I am interested in why `InterlockedExchange()` is being used here (and below) when setting the value for a struct which (as far as I can see) will only be called from the main thread, once, during startup.

Reply:
restore_terminal can be called from the signal handler which the system calls from a separate thread

Reply:
Ah, got it. Thanks. Didn't realize that.